### ISSUE TITLE
False positive on lombok @With and generated methods

### ISSUE THREAD

**Issue by willypm33** [spotbugs/sonar-findbugs#1164]
SonarQube server Developer Edition v2025.1 (102418)
sonar-findbugs v4.4.0

For **With** annotation, we have the rule ID **findbugs:RC_REF_COMPARISON** on the withxxx generated methods by Lombok.

```
import lombok.With;

@With
public record Instance(
    Integer order
) {

}
```

**Comment by gtoison** [comment]
So this should be fixed with version [4.4.2](https://github.com/spotbugs/sonar-findbugs/releases/tag/4.4.2) of the plugin
